pkg/lwip/contrib/netif/lwip_netif.c

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (C) 2018 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ *
+ * @file
+ * @author  Martine Lenders <m.lenders@fu-berlin.de>
+ */
+
+#include <stdint.h>
+
+#include "fmt.h"
+#include "net/eui64.h"
+#include "net/netdev.h"
+
+#include "lwip/netif.h"
+#include "lwip/opt.h"
+#include "netif/lowpan6_opts.h"
+
+#include "netif_types.h"
+
+netif_t netif_iter(netif_t last)
+{
+    if (last == NETIF_INVALID) {
+        return netif_list;
+    }
+    else {
+        return last->next;
+    }
+}
+
+int netif_get_name(netif_t netif, char *name)
+{
+    size_t offset = sizeof(netif->name);
+
+    /* construct name expected by lwIP's `netif_find()` */
+    memcpy(name, netif->name, offset);
+    offset += fmt_u16_dec(&name[offset], netif->num);
+    name[offset] = '\0';
+    return offset;
+}
+
+netif_t netif_get_by_name(const char *name)
+{
+    return netif_find(name);
+}
+
+int netif_get_opt(netif_t netif, netopt_t opt, uint16_t context,
+                  void *value, size_t max_len)
+{
+    netdev_t *dev = netif->state;
+
+    (void)context;
+    switch (opt) {
+#if LWIP_IPV6
+        case NETOPT_IPV6_IID:
+            return sizeof(eui64_t);
```

Comment:
same as for the `set_opt`, I don't understand this - and I also think this is wrong because the docu says:

```
 * @return  Number of bytes written to @p value.
```

but here nothing is written to `value`, or did I miss something?!

Reply:
and if this assumes `value` is `0`: I would rather not count on that, and explicitly set the default value.

Reply:
> This is the WIP part of this PR ;-) the proper functions will be called here later.

Dito